docs/integrations/gitlab/discovery.md

```diff
@@ -59,19 +120,45 @@ builder.addEntityProvider(
     }),
     // optional: alternatively, use schedule
     scheduler: env.scheduler,
-  }),
-);
+  });
+  env.eventBroker.subscribe(gitlabProvider);
+  builder.addEntityProvider(gitlabProvider);
+  /* highlight-add-end */
+  const { processingEngine, router } = await builder.build();
+  await processingEngine.start();
+  return router;
+}
 ```
 
-## Alternative processor
+## Configuration
 
-```yaml
+To use the discovery provider, you'll need a GitLab integration
+[set up](locations.md) with a `token`. Then you can add a provider config per group
+to the catalog configuration.
+
+> > NOTE: if you are using the New Backend System, the `schedule` has to be setup in the config, as shown below.
+
+```yaml title="app-config.yaml"
 catalog:
-  locations:
-    - type: gitlab-discovery
-      target: https://gitlab.com/group/subgroup/blob/main/catalog-info.yaml
+  providers:
+    gitlab:
+      yourProviderId:
+        host: gitlab-host # Identifies one of the hosts set up in the integrations
+        branch: main # Optional. Used to discover on a specific branch
+        fallbackBranch: master # Optional. Fallback to be used if there is no default branch configured at the Gitlab repository. It is only used, if `branch` is undefined. Uses `master` as default
+        skipForkedRepos: false # Optional. If the project is a fork, skip repository
+        group: example-group # Optional. Group and subgroup (if needed) to look for repositories. If not present the whole instance will be scanned
+        entityFilename: catalog-info.yaml # Optional. Defaults to `catalog-info.yaml`
+        projectPattern: '[\s\S]*' # Optional. Filters found projects based on provided patter. Defaults to `[\s\S]*`, which means to not filter anything
+        schedule: # optional; same options as in TaskScheduleDefinition
```

Comment:
Is this true in all cases that it's optional? If I'm using the New Backend System would that not then be required?

Reply:
Changed this line as well!